src/compute/src/compute_state.rs

```diff
@@ -165,33 +167,37 @@ impl<'a, A: Allocate, B: ComputeReplay> ActiveComputeState<'a, A, B> {
             }
 
             ComputeCommand::Peek(peek) => {
-                // Acquire a copy of the trace suitable for fulfilling the peek.
-                let mut trace_bundle = self.compute_state.traces.get(&peek.id).unwrap().clone();
-                let timestamp_frontier = Antichain::from_elem(peek.timestamp);
-                let empty_frontier = Antichain::new();
-                trace_bundle
-                    .oks_mut()
-                    .set_logical_compaction(timestamp_frontier.borrow());
-                trace_bundle
-                    .errs_mut()
-                    .set_logical_compaction(timestamp_frontier.borrow());
-                trace_bundle
-                    .oks_mut()
-                    .set_physical_compaction(empty_frontier.borrow());
-                trace_bundle
-                    .errs_mut()
-                    .set_physical_compaction(empty_frontier.borrow());
-                // Prepare a description of the peek work to do.
-                let mut peek = PendingPeek { peek, trace_bundle };
-                // Log the receipt of the peek.
-                if let Some(logger) = self.compute_state.materialized_logger.as_mut() {
-                    logger.log(ComputeEvent::Peek(peek.as_log_event(), true));
-                }
-                // Attempt to fulfill the peek.
-                if let Some(response) = peek.seek_fulfillment(&mut Antichain::new()) {
-                    self.send_peek_response(peek, response);
-                } else {
-                    self.compute_state.pending_peeks.push(peek);
+                // Only handle peeks that are not targeted at a different replica.
+                let target = peek.target_replica;
+                if target.is_none() || target == Some(self.compute_state.replica_id) {
+                    // Acquire a copy of the trace suitable for fulfilling the peek.
+                    let mut trace_bundle = self.compute_state.traces.get(&peek.id).unwrap().clone();
+                    let timestamp_frontier = Antichain::from_elem(peek.timestamp);
+                    let empty_frontier = Antichain::new();
+                    trace_bundle
+                        .oks_mut()
+                        .set_logical_compaction(timestamp_frontier.borrow());
+                    trace_bundle
+                        .errs_mut()
+                        .set_logical_compaction(timestamp_frontier.borrow());
+                    trace_bundle
+                        .oks_mut()
+                        .set_physical_compaction(empty_frontier.borrow());
+                    trace_bundle
+                        .errs_mut()
+                        .set_physical_compaction(empty_frontier.borrow());
```

Comment:
I think this is fine to only do here since `AllowCompaction` will advance the frontiers independently and we don't actually rely on `Peek` to advance anything.

Reply:
If I understand this correctly, the only reason we do these set compaction calls is that we want to prevent pending peeks from holding back compaction more than absolutely necessary. If we don't create the `trace_bundle` clone in the first place, we don't have to worry about that.